storage/storage.go

```diff
@@ -1386,6 +1418,19 @@ type ObjectAttrs struct {
 	// Retention contains the retention configuration for this object.
 	// ObjectRetention cannot be configured or reported through the gRPC API.
 	Retention *ObjectRetention
+
+	// SoftDeleteTime is the time when the object became soft-deleted.
+	// Soft-deleted objects are only accessible on an object handle returned by
```

Comment:
What happens when you do `ObjectHandle.Attrs()` on an object that is soft deleted without setting `ObjectHandle.SoftDeleted`?